internal/merkle/clock/clock.go

```diff
@@ -221,6 +218,49 @@ func encryptBlock(
 	return &coreblock.Block{Delta: clonedCRDT, Heads: block.Heads, Links: block.Links}, nil
 }
 
+func (mc *MerkleClock) signBlock(
+	ctx context.Context,
+	block *coreblock.Block,
+) error {
+	// We sign only the first field blocks just to add entropy and prevent any collisions.
+	// The integrity of the field data is guaranteed by signatures of the parent composite blocks.
+	if block.Delta.IsField() && block.Delta.GetPriority() > 1 {
+		return nil
+	}
+
+	ident := identity.FromContext(ctx)
+	if !ident.HasValue() {
+		return nil
+	}
+
+	blockBytes, err := block.Marshal()
+	if err != nil {
+		return err
+	}
+
+	sigBytes, err := crypto.Sign(crypto.SignatureTypeECDSA256K, ident.Value().PrivateKey, blockBytes)
+	if err != nil {
+		return err
+	}
+
+	sig := &coreblock.Signature{
+		Header: coreblock.SignatureHeader{
+			Type:     coreblock.SignatureTypeECDSA256K,
+			Identity: []byte(ident.Value().PublicKey.SerializeCompressed()),
+		},
+		Value: sigBytes,
+	}
+
+	sigBlockLink, err := putBlock(ctx, mc.blockstore, sig)
```

Comment:
question: Out of curiosity... You removed `putBlock` and `putEncBlock` from being a method of `MerkleClock`, my assumption is because they only relied on a single field from the `MerkleClock` type (specifically `blockstore`), but then added `signBlock` as a method to `MerkleClock` even though it also only relies on the one `blockstore` field.

Reply:
discussed: ignore this, I'm stupid. Missed a type parameter in the function definition.